electrum/transaction.py

```diff
@@ -1331,6 +1349,19 @@ def scriptpubkey(self) -> Optional[bytes]:
             return self.witness_utxo.scriptpubkey
         return None
 
+    def set_script_type(self) -> None:
+        if self.scriptpubkey is None:
+            return
+        type = get_script_type_from_output_script(self.scriptpubkey)
+        if type is not None:
+            if type in ('p2sh', 'p2wsh'):
+                if self.redeem_script is not None:
```

Comment:
do you expect `type == 'p2wsh'`?
note that if `type == 'p2wsh'`, it is `self.witness_script` that should be checked
(if `type == 'p2sh'` then it is `self.redeem_script`)

Reply:
changed